synapse/handlers/message.py

```diff
@@ -1159,6 +1159,9 @@ def is_inviter_member_event(e):
                 if original_event.room_id != event.room_id:
                     raise SynapseError(400, "Cannot redact event from a different room")
```

Comment:
hmm, this should probably be a 403, alas. out of scope for this PR though.

Reply:
No, this is a 400; redacting a room's event using a different room ID is ill-formed, it is not "well-formed but forbidden".